c8run/internal/archive/archive.go

```diff
@@ -0,0 +1,289 @@
+package archive
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func DownloadFile(filepath string, url string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
```

Comment:
You could include context in error messages to make them more informative. For example `fmt.Errorf("...")`

Reply:
I like the idea, but in implementation, my first attempt at this resulted in an error message like 

"Failed to create file named file.txt, Original Error: file.txt already exists"

Which isn't any more descriptive than "file.txt already exists".

What sort of context do people usually add to their error messages? do they append filenames+line numbers ?

Reply:
Usually its "DownloadFile: error on file creation: <error coming from os.Create>" but it depends a bit what you want to expose to the consumer of your function. 

The text usually helps to identify the code path that failed by searching for the string. Some logging libraries also support stack traces. 

Go Blog about wrapping errors: https://go.dev/blog/go1.13-errors#errors-and-package-apis

Reply:
I included function name  prefixes to each of the errors in d82b4686346 as you say.  Also included better log messages for some of the functions.